Utils/Dataflow/095_datasetInfoAMI/amiDatasets.py

```diff
@@ -139,7 +154,14 @@ def process(stage, message):
     # 'data_format' list contains one of the allowed formats
     # or not set at all.
     if update or not formats:
-        amiPhysValues(data)
+        try:
+            amiPhysValues(data)
+        except DataflowException:
+            raise
+        except Exception:
+            stage.output_error("Failed to process dataset '%s'"
+                               % data['datasetname'], sys.exc_info())
+        change_key_names(data)
```

Comment:
This is, indeed, the result of

> "as things were before"

But now, when you brought this point up... What can cause exception in `change_key_names()`?

1. `PHYS_VALUES` is not defined.
2. There is no `'ami'` in some `item` - which means that `PHYS_VALUES` was defined incorrectly.
3. Same with `'es'`.
4. `data` is not a `dict`.

First three points do not depend on `data` - if we run into them, then there is a problem with the stage, which means that it is logical to stop after the first instance [1], not process each dataset (while putting pressure on AMI) and then throw it away with "I can't process this!". The fourth one depends on `data`, but there is (as you pointed out) a check in `process()`, and if `process()` makes the check and then transforms `data` into something different... Again, looks like a problem with the stage that calls for stop.

So, unless I am missing something, the function turned out to be in the right place, even if it wasn't really intentional. It is possible for the function to become more complex in the future, where maybe there would be actual problems with given `data`, and then moving it into `try...except` would be reasonable... But not now.

[1] If we really care about this, then it would be better to check that `PHYS_VALUES` is defined, and defined properly, somewhere in the beginning, not during the processing of each dataset. But we do not.

Reply:
I made this digging into the `change_key_names()` as well, and -- technically -- you're right.

My point here is that for now it is not clear from the code if author(s) thought about this moment, or it just "historically happened" -- which makes others like me (who always have doubts even about self-written code, not to mention someone else's one) stumble here and try to find out if there's a possible misfunction, or everything's (accidentally ;) ) fine.

It could be done more clear by something like this:

```
try:
    change_key_names(data)
except Exception, err:
    raise DataflowException("Stage internal failure.", reason=err)
```

However (I didn't check) most likely it will mask the original `err` and makes it harder to find out the reason... so maybe just point out that it is not an accident:

```
try:
   ...
except:
   ...

# Do not try/except this one -- any exception would indicate an internal
# functionality problem, meaning that the stage cannot be used until it's fixed
change_key_names(data)

stage.output(...)
return True
```

My version may be too wordy (even being much shorter than the first version ;) ), but it's just an idea -- if it makes any sense, it can be shortened/refined to your liking.

Reply:
Added a comment.